base/error.jl

```diff
@@ -18,24 +18,77 @@
 
 ## native julia error handling ##
 
+"""
+    error(message::AbstractString)
+
+Raise an `ErrorException` with the given message.
+"""
 error(s::AbstractString) = throw(ErrorException(s))
 error(s...) = throw(ErrorException(Main.Base.string(s...)))
 
+"""
+    rethrow([e])
+
+Throw an object without changing the current exception backtrace. The default argument is
+the current exception (if called within a `catch` block).
+"""
 rethrow() = ccall(:jl_rethrow, Bottom, ())
 rethrow(e) = ccall(:jl_rethrow_other, Bottom, (Any,), e)
+
+"""
+    backtrace()
+
+Get a backtrace object for the current program point.
+"""
 backtrace() = ccall(:jl_backtrace_from_here, Array{Ptr{Void},1}, (Int32,), false)
+
+"""
+    catch_backtrace()
+
+Get the backtrace of the current exception, for use within `catch` blocks.
+The backtrace is represented as an Array{Ptr{Void}} of addresses, so [`showerror`](:func:`showerror`)
+may be helpful for displaying the backtrace in a human-readable format.
+
+# Example
+
+```
+f(x) = try
+    sqrt(x)
+catch y
+    bt = catch_backtrace()
+    showerror(STDOUT, y, bt)
+end
+```
```

Comment:
The output seems system dependent. I get

`DomainError:
 in f(::Int64) at ./REPL[49]:2
 in eval(::Module, ::Any) at ./boot.jl:238
 in eval(::Module, ::Any) at /Users/jane/Applications/julia/usr/lib/julia/sys.dylib:?
 in eval_user_input(::Any, ::Base.REPL.REPLBackend) at ./REPL.jl:66
 in macro expansion at ./REPL.jl:97 [inlined]
 in (::Base.REPL.##3#4{Base.REPL.REPLBackend})() at ./event.jl:65`

Is there a work-around/way to exclude local paths so I can include a doctest?